test/c3/appframework/appframework_test.go

```diff
@@ -514,7 +569,132 @@ var _ = Describe("c3appfw test", func() {
+			// Ensure Indexers go to Ready phase
+			testenv.SingleSiteIndexersReady(deployment, testenvInstance)
+
+			// Ensure SHC go to Ready phase
+			testenv.SearchHeadClusterReady(deployment, testenvInstance)
+
+			// Verify RF SF is met
+			testenv.VerifyRFSFMet(deployment, testenvInstance)
+
+			// Verify apps with local scope are downloaded by init-container
+			initContDownloadLocation = "/init-apps/" + appSourceNameLocal
+			podNames = []string{fmt.Sprintf(testenv.ClusterMasterPod, deployment.GetName()), fmt.Sprintf(testenv.DeployerPod, deployment.GetName())}
+			appFileList = testenv.GetAppFileList(appListLocal, 2)
+			testenv.VerifyAppsDownloadedByInitContainer(deployment, testenvInstance, testenvInstance.GetName(), podNames, appFileList, initContDownloadLocation)
+
+			// Verify apps with cluster scope are downloaded by init-container
+			initContDownloadLocation = "/init-apps/" + appSourceNameCluster
+			podNames = []string{fmt.Sprintf(testenv.ClusterMasterPod, deployment.GetName()), fmt.Sprintf(testenv.DeployerPod, deployment.GetName())}
+			appFileList = testenv.GetAppFileList(appListCluster, 2)
+			testenv.VerifyAppsDownloadedByInitContainer(deployment, testenvInstance, testenvInstance.GetName(), podNames, appFileList, initContDownloadLocation)
+
+			// Verify apps with local scope are installed locally on CM and on SHC Deployer
+			testenv.VerifyAppInstalled(deployment, testenvInstance, testenvInstance.GetName(), podNames, appListLocal, true, "enabled", true, false)
+
+			// Verify apps with cluster scope are installed on indexers
+			podNames = []string{}
+			for i := 0; i < int(indexerReplicas); i++ {
+				sh := fmt.Sprintf(testenv.IndexerPod, deployment.GetName(), i)
+				podNames = append(podNames, string(sh))
+			}
+			testenv.VerifyAppInstalled(deployment, testenvInstance, testenvInstance.GetName(), podNames, appListCluster, true, "enabled", true, true)
+
+			// Verify apps with cluster scope are installed on SHs
+			podNames = []string{}
+			for i := 0; i < int(shReplicas); i++ {
+				sh := fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), i)
+				podNames = append(podNames, string(sh))
+			}
+			testenv.VerifyAppInstalled(deployment, testenvInstance, testenvInstance.GetName(), podNames, appListCluster, true, "enabled", true, true)
+
+			//Delete apps on S3 for new Apps
+			testenv.DeleteFilesOnS3(testS3Bucket, uploadedApps)
+			uploadedApps = nil
+			testenvInstance.Log.Info("Testing downgrade scenario")
+
+			//Upload new Versioned Apps to S3 to test downgrade scenario
+			appFileList = testenv.GetAppFileList(appListLocal, 1)
+			uploadedFiles, err = testenv.UploadFilesToS3(testS3Bucket, s3TestDir, appFileList, downloadDirV1)
+			Expect(err).To(Succeed(), "Unable to upload apps to S3 test directory")
+			uploadedApps = append(uploadedApps, uploadedFiles...)
+
+			// Upload apps to a 2nd S3 bucket as we need 2 buckets for this test.(appListLocal apps to be used for local install, appListCluster apps for cluster install)
```

Comment:
Nit: Good idea to modify this comment as the it's not 2nd S3 bucket but just a different directory on same bucket.

Reply:
Modified as '2nd Directory on S3 bucket'